packages/docs/developer-resources/contractkit/odis.md

```diff
@@ -0,0 +1,113 @@
+# Calling Oblivious Decentralized Identifier Service (ODIS)
+This guide walks through using Contract Kit to query the on-chain identifier given a phone number or other identifier (ex. Twitter handle). See [this overview document](../../celo-codebase/protocol/identity/phone-number-privacy.md) for more details on ODIS. One of Celo's key features is the ability to tie a phone number to a Celo address. This provides a convenient payment experience for Celo users. To map a phone number to an address, the on-chain identifier for a given phone number must first be retrieved. With this identifier, the address can be looked up on-chain.
+
+{% hint style="info" %}
+ODIS requests are rate-limited based on transaction fees. Ensure the account that is performing the queries has a balance and has performed transactions on the network. If an out of quota error is hit, this indicates that more transactions need to be performed.
+{% endhint %}
+
+There are two methods for ODIS:
+1. `getPhoneNumberIdentifier` - Query and compute the identifier for a phone number
+2. `getContactMatches` - Find mutual connections between users
+
+## Authentication
+Both methods require authentication to the ODIS server, which can be performed by either the main wallet key or the data-encryption key (DEK) associated with the wallet key. This is managed by `AuthSigner`, which can be either a `WalletKeySigner` for a wallet key or an `EncryptionKeySigner` for the DEK. The DEK method is preferred, since it doesn't require the user to access the same key that manages their funds. Before using the DEK, it's important to check that the DEK is the latest that's registered for a user. 
+```ts
+  // Query the on-chain data encryption key for a user
+  const accountWrapper: AccountsWrapper = await contractKit.contracts.getAccounts()
+  const dataEncryptionKey = accountWrapper.getDataEncryptionKey(address)
```

Comment:
I think this snippet may be misleading. It may lead people to think they can get the DEK private key from the contract. Should we cut it and assume that queries already have some solution for DEK management? A link to a doc about the DEK may do more good here.

Reply:
Good suggestion. I'll create a new DEK doc and link it.